python-sdk/src/astro/databases/mssql.py

```diff
@@ -0,0 +1,378 @@
+        Currently, Schema is not being fetched from airflow connection for Mssql because, in Mssql,
+        databases and schema are different concepts
+        The MssqlHook only exposes schema
+        However, implementation-wise, it seems that if the MssqlHook receives a schema during
+        initialization, but it uses it as a database in the connection to Mssql:
+        """
+        # TODO: Change airflow MssqlHook to fetch database and schema separately
+        database = self.hook.get_connection(self.conn_id).schema
+        return Metadata(database=database, schema=self.DEFAULT_SCHEMA)  # type: ignore
+
+    def schema_exists(self, schema) -> bool:
+        """
+        Checks if a schema exists in the database
+        :param schema: DB Schema - a namespace that contains named objects like (tables, functions, etc)
+        """
+        schema_result = self.hook.run(
+            "SELECT schema_name FROM information_schema.schemata WHERE lower(schema_name) = lower(%(schema_name)s);",
+            parameters={"schema_name": schema.lower()},
+            handler=lambda x: [y[0] for y in x.fetchall()],
+        )
+        return len(schema_result) > 0
+
+    def run_sql(
+        self,
+        sql: str | ClauseElement = "",
+        parameters: dict | None = None,
+        autocommit: bool = False,
+        **kwargs,
+    ) -> CursorResult:
+        """
+        Return the results to running a SQL statement.
+        Whenever possible, this method should be implemented using Airflow Hooks,
+        since this will simplify the integration with Async operators.
+        :param sql: Contains SQL query to be run against database
+        :param parameters: Optional parameters to be used to render the query
+        :param autocommit: specify autocommit setting for the query
+        """
+        if parameters is None:
+            parameters = {}
+
+        if "sql_statement" in kwargs:  # pragma: no cover
+            warnings.warn(
+                "`sql_statement` is deprecated and will be removed in future release"
+                "Please use  `sql` param instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+            sql = kwargs.get("sql_statement")  # type: ignore
+
+        # We need to set autocommit=True for specific queries
+        if isinstance(sql, str):
+            if (autocommit is True) or any(
+                query in sql.lower() for query in self._queries_requiring_autocommit
+            ):
+                self.connection.execute(sqlalchemy.text(sql).execution_options(autocommit=True), parameters)
+            else:
+                result = self.connection.execute(
+                    sqlalchemy.text(sql).execution_options(autocommit=autocommit), parameters
+                )
+                return result
+        else:
+            # this is used for append
+            result = self.connection.execute(sql, parameters)
```

Comment:
@phanikumv We don't need auto-commit options when the query is not string?

Reply:
no we don't as its a compiled sql statement that comes into the else block